runtime/queries/metricsview_rows.go

```diff
@@ -91,6 +94,45 @@ func (q *MetricsViewRows) Resolve(ctx context.Context, rt *runtime.Runtime, inst
 	return nil
 }
 
+func (q *MetricsViewRows) Export(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int, format runtimev1.DownloadFormat, writer io.Writer) error {
+	err := q.Resolve(ctx, rt, instanceID, priority)
+	if err != nil {
+		return err
+	}
+
+	switch format {
+	case runtimev1.DownloadFormat_DOWNLOAD_FORMAT_UNSPECIFIED, runtimev1.DownloadFormat_DOWNLOAD_FORMAT_CSV:
+		return writeCSV(q.Result.Meta, q.Result.Data, writer)
+	}
+
+	return nil
+}
+
+func writeCSV(meta []*runtimev1.MetricsViewColumn, data []*structpb.Struct, writer io.Writer) error {
+	w := csv.NewWriter(writer)
+
+	record := make([]string, 0, len(meta))
+	for _, field := range meta {
+		record = append(record, field.Name)
+	}
+	if err := w.Write(record); err != nil {
+		return err
+	}
+	record = record[:0]
+
+	for _, structs := range data {
+		for _, field := range structs.Fields {
+			record = append(record, field.GetStringValue())
+		}
```

Comment:
`GetStringValue()` will return nil if the field doesn't have a string type. I think `AsInterface()` or `String()` are what you need

Reply:
Also, are you certain that slicing to 0 and then calling `append` will re-use the same array? I know slicing keeps the underlying array, but couldn't immediately find info about whether `append` includes safety measures to prevent overriding underlying data that can be referenced by other slices.

I think it should work as expected, just want to be sure